kmath-for-real/src/commonTest/kotlin/kaceince/kmath/real/RealVectorTest.kt

```diff
@@ -32,7 +30,7 @@ internal class RealVectorTest {
         val vector2 =  Buffer.real(5) { 5 - it.toDouble() }
         val matrix1 = vector1.asMatrix()
         val matrix2 = vector2.asMatrix().transpose()
-        val product = MatrixContext.real { matrix1 dot matrix2 }
+        val product = LinearSpace.real.run { matrix1 dot matrix2 }
```

Comment:
I think we need to develop a certain convention to choose scoping functions.

Reply:
Indeed. `invoke` seems to work badly because of import and it is not possible to make it a member because of self-types. I think we should use `with` everywhere.